src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructorParameterBackingFieldSymbol.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a compiler generated backing field for a primary constructor parameter.
+    /// </summary>
+    internal sealed class SynthesizedPrimaryConstructorParameterBackingFieldSymbol : SynthesizedBackingFieldSymbolBase
+    {
+        public readonly ParameterSymbol ParameterSymbol;
+
+        public SynthesizedPrimaryConstructorParameterBackingFieldSymbol(
+            ParameterSymbol parameterSymbol,
+            string name)
+            : base(name, isReadOnly: false, isStatic: false)
+        {
+            ParameterSymbol = parameterSymbol;
+        }
+
+        internal override bool HasInitializer => true;
+
+        protected override IAttributeTargetSymbol AttributeOwner
+            => this;
+
+        internal override Location ErrorLocation
+            => ParameterSymbol.Locations.FirstOrDefault() ?? NoLocation.Singleton;
+
+        protected override SyntaxList<AttributeListSyntax> AttributeDeclarationSyntaxList
+            => default;
+
+        public override Symbol? AssociatedSymbol
```

Comment:
>AssociatedSymbol [](http://example.com/codeflow?start=32&length=16)

Should this return the `ParameterSymbol`? The documentation says "If this field serves as a backing variable for an automatically generated property or a field-like event, returns that property/event. Otherwise returns null. Note, the set of possible associated symbols might be expanded in the future to reflect changes in the languages."

Reply:
> Should this return the ParameterSymbol? 

> Note, the set of possible associated symbols might be expanded in the future to reflect changes in the languages.

The set might be expanded, but doesn't have to be expanded. We will do that once we will have a compelling scenario. At the moment we do not have one.